lua/nvim-tree/diagnostics.lua

```diff
@@ -114,7 +114,11 @@ function M.update()
         for line, node in pairs(nodes_by_line) do
           local nodepath = utils.canonical_path(node.absolute_path)
           log.line("diagnostics", "  %d checking nodepath '%s'", line, nodepath)
-          if M.show_on_dirs and vim.startswith(bufpath, nodepath) and (not node.open or M.show_on_open_dirs) then
+          if
+            M.show_on_dirs
+            and vim.startswith(bufpath:gsub("\\", "/"), nodepath:gsub("\\", "/") .. "/")
```

Comment:
Good catch!

Unfortunately we need to deal with windows powershell paths `C:\User\me`

Rather than messing around with path separators could we use something like `vim.fn.fnamemodify` `:h` and match on that?

string.find plain is also useful so that the separators aren't treated as magic characters.

Reply:
The problem with `:h` is that `vim.fn.fnamemodify('lua/nvim-tree/diagnostics.lua', ':h')` will match `'lua/nvim'`. Though I do agree that `fnamemodify` is better and cleaner. I would be using `:p` instead which will add `/` if path is a directory.